client/src/context/GameContext.tsx

```diff
@@ -190,13 +194,32 @@ export const GameProvider: React.FC = ({ children }) => {
     }
   }
 
+  function restartLevel() {
+    setStepIndex(0);
+    setAttempts(0);
+    setValues({});
+    setCorrect({});
+  }
+
+  function hasSeenLevel(lastLevel: number) {
+      if (lastLevel < maxLevelSeen) {
+          return false;
+      }
+      return true;
+    }
+
```

Comment:
So really, you have "seen" the `lastLevel` when `lastLevel < maxLevelSeen`, not the other way around as is defined right now. So really it should return `true` when `lastLevel < maxLevelSeen`, however, this whole method may be simplified to `return lastLevel < maxLevelSeen`.